src/contrib/inference/dynamichmc.jl

```diff
@@ -61,8 +61,15 @@ function AbstractMCMC.sample_init!(
     end
 
     model(spl.state.vi, SampleFromUniform())
+    link!(spl.state.vi, spl)
```

Comment:
why do we add `link!` here @mohamed82008

Reply:
Because `SampleFromUniform` calls `invlink` https://github.com/TuringLang/DynamicPPL.jl/blob/master/src/utils.jl#L131 and sets `trans` to false. Whether we want that or not is a separate issue. But for now that's the case. I changed it for a brief moment in DPPL but it was changed back.

Reply:
I see. Thanks for confirming.

Reply:
Great thanks for confirming.